Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerInfo.cs

```diff
@@ -56,14 +59,94 @@ public class MotionControllerInfo
+            Touchpad,
+            // Controller body elements & poses
+            PointingPose
+        }
+
+        public bool TryGetElement(ControllerElementEnum element, out Transform elementTransform)
+        {
+            switch (element)
+            {
+                // control elements
+                case ControllerElementEnum.Home:
+                    if (home != null)
+                    {
+                        elementTransform = home.transform;
+                        return true;
+                    }
+                    break;
+                case ControllerElementEnum.Menu:
+                    if (menu != null)
+                    {
+                        elementTransform = menu.transform;
+                        return true;
+                    }
+                    break;
+                case ControllerElementEnum.Select:
+                    if (select != null)
+                    {
+                        elementTransform = select.transform;
+                        return true;
+                    }
+                    break;
+                case ControllerElementEnum.Grasp:
+                    if (grasp != null)
+                    {
+                        elementTransform = grasp.transform;
+                        return true;
+                    }
+                    break;
+                case ControllerElementEnum.Thumbstick:
+                    if (thumbstickPress != null)
+                    {
+                        elementTransform = thumbstickPress.transform;
+                        return true;
+                    }
+                    break;
+                case ControllerElementEnum.Touchpad:
+                    if (touchpadPress != null)
+                    {
+                        elementTransform = touchpadPress.transform;
+                        return true;
+                    }
+                    break;
+                // body elements & poses
+                case ControllerElementEnum.PointingPose:
+                    if (pointingPose != null)
+                    {
+                        elementTransform = pointingPose.transform;
+                        return true;
+                    }
+                    break;
+            }
+
+            elementTransform = null;
+            return false;
+        }
+
         /// <summary>
         /// Iterates through the Transform array to find specifically named GameObjects.
         /// These GameObjects specify the animation bounds and the GameObject to modify for button,
         /// thumbstick, and touchpad animation.
         /// </summary>
         /// <param name="childTransforms">The transforms of the glTF model.</param>
         /// <param name="visualizerScript">The script containing references to any objects to spawn.</param>
-        public void LoadInfo(Transform[] childTransforms, MotionControllerVisualizer visualizerScript)
+        public void LoadInfo(Transform[] childTransforms)
```

Comment:
breaking change?

should we overload and add obsolete message?

Reply:
Since controller support is so new and nobody's dug in yet I feel an outright breaking change would make the most sense. Just rip the band-aid off so devs know what's what.